### PR TITLE
[Core][LockObject] deleting Copy- & MoveCtor and Assignment Operator

### PR DESCRIPTION
**Pull request by philbucher**
Reason is that it is not clear if the omp-lock is allowed to be moved, copied or assigned => might be implementation dependent and only fail for certain compilers

TODO:
- [ ] @pooyan-dadvand still needs to remove the usage from `Chunk`

### CODE REVIEW

File: kratos/includes/lock_object.h [hunk 1]
```diff
@@ -48,23 +48,16 @@ namespace Kratos
...
 
 		/// Move constructor.
-		LockObject(LockObject&& rOther) noexcept
+    LockObject(LockObject&& rOther) noexcept
 #ifdef KRATOS_SMP_OPENMP
 			: mLock(rOther.mLock)
 #endif
 		{
 #ifdef KRATOS_SMP_OPENMP
+      static_assert(std::is_move_constructible<omp_lock_t>::value, "omp_lock_t is not move constructible!");
```
Comment (philbucher) [inline kratos/includes/lock_object.h:60]: see https://en.cppreference.com/w/cpp/types/is_move_constructible

**Comment by philbucher** [comment]
@pooyan-dadvand at least with GCC is compiles, lets see what the other compilers say

**Comment by philbucher** [comment]
ok seems to works fine for the other compilers too

**Review by pooyan-dadvand — APPROVED** [review]
Thanks!

**Pull request merged**